### ISSUE TITLE
Startup issues

### ISSUE THREAD

**Issue by chabanachtaTU** [borkdominik/bigUML#170]
Hi, 

Sorry this maybe a stupid question and just a user error. I am trying out the extension, and getting the following output when creating a new file or opening an existing .uml file (vscode on windows 11, openjdk 21.0.2 and 23.0.1), and it hangs indefinitely with the alert "Model loading in progress ...". So there seems to be some "Internal server error" string when json is expected.
________________________________________________________________

`Registered server launchers: ModelServer, GLSPServer
Enabled server launchers: ModelServer, GLSPServer
The [Model-Server] listens on port 58657
[GLSP-Server] 2024-10-26 13:02:00,593037100 [main] INFO  SocketGLSPServerLauncher - The GLSP server is ready to accept new client requests on port: 58656

The [GLSP-Server] listens on port 58656
[GLSP-Server] [GLSP-Server]:Startup completed. Accepting requests on port:58656

[GLSP-Server] 2024-10-26 13:02:13,630362600 [DefaultActionDispatcher 1] INFO  DefaultActionDispatcher - Terminating DefaultActionDispatcher thread DefaultActionDispatcher 1
2024-10-26 13:02:13,630362600 [DefaultActionDispatcher 1] INFO  DefaultActionDispatcher - Terminating DefaultActionDispatcher

[GLSP-Server] 2024-10-26 13:02:13,654241100 [Thread-17] INFO  SocketGLSPServerLauncher - Starting GLSP server connection for client /127.0.0.1:58661

[GLSP-Server] 2024-10-26 13:02:13,676233200 [Thread-17] INFO  SocketGLSPServerLauncher - Stopping GLSP server connection for client /127.0.0.1:58661
2024-10-26 13:02:13,676233200 [Thread-17] INFO  DefaultGLSPServer - Shutdown GLSP Server com.eclipsesource.uml.glsp.core.UmlGLSPServer@2ca42d15

[GLSP-Server] 2024-10-26 13:02:14,175570000 [DefaultActionDispatcher 3] INFO  DefaultActionDispatcher - Terminating DefaultActionDispatcher thread DefaultActionDispatcher 3
2024-10-26 13:02:14,175570000 [DefaultActionDispatcher 3] INFO  DefaultActionDispatcher - Terminating DefaultActionDispatcher

[GLSP-Server] 2024-10-26 13:02:14,177569400 [Thread-17] INFO  SocketGLSPServerLauncher - Starting GLSP server connection for client /127.0.0.1:58670

[GLSP-Server] 2024-10-26 13:02:15,653007400 [OkHttp http://localhost:58657/...] ERROR ModelServerClientDelegate - Could not parse JSON
com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'Internal': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (String)"Internal server error"; line: 1, column: 9]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2391) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:745) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:2961) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:2939) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._matchToken(ReaderBasedJsonParser.java:2713) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:1986) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:802) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper._readTreeAndClose(ObjectMapper.java:4703) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3076) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.emfcloud.modelserver.jsonschema.Json.parse(Json.java:34) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.emfcloud.modelserver.internal.client.ModelServerClientDelegate.parseJsonField(ModelServerClientDelegate.java:831) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.emfcloud.modelserver.internal.client.ModelServerClientDelegate.lambda$35(ModelServerClientDelegate.java:826) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at java.util.function.Function.lambda$andThen$1(Function.java:88) ~[?:?]
	at java.util.function.Function.lambda$andThen$1(Function.java:88) ~[?:?]
	at java.util.function.Function.lambda$andThen$1(Function.java:88) ~[?:?]
	at org.eclipse.emfcloud.modelserver.client.Response.body(Response.java:41) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.emfcloud.modelserver.internal.client.ModelServerClientDelegate.lambda$6(ModelServerClientDelegate.java:205) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179) ~[?:?]
	at org.eclipse.emfcloud.modelserver.internal.client.ModelServerClientDelegate$3.onResponse(ModelServerClientDelegate.java:818) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.lang.Thread.run(Thread.java:1583) ~[?:?]

[GLSP-Server] 2024-10-26 13:02:15,683002400 [DefaultActionDispatcher 5] ERROR UmlModelServerAccess - java.util.concurrent.ExecutionException: java.lang.RuntimeException: Could not parse 'data' field

[GLSP-Server] 2024-10-26 13:02:15,684000700 [DefaultActionDispatcher 5] ERROR DefaultGLSPServer - [ERROR] Could not process message:ActionMessage [action=requestModel, clientId=umldiagram_0]
org.eclipse.glsp.server.types.GLSPServerException: Error during model loading
	at com.eclipsesource.uml.glsp.core.model.UmlModelServerAccess.getSemanticModel(UmlModelServerAccess.java:54) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.emfcloud.modelserver.glsp.notation.integration.EMSNotationSourceModelStorage.doLoadSourceModel(EMSNotationSourceModelStorage.java:29) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.emfcloud.modelserver.glsp.EMSSourceModelStorage.loadSourceModel(EMSSourceModelStorage.java:33) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.glsp.server.features.core.model.RequestModelActionHandler.executeAction(RequestModelActionHandler.java:75) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.emfcloud.modelserver.glsp.actions.handlers.EMSRequestModelActionHandler.executeAction(EMSRequestModelActionHandler.java:31) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.glsp.server.features.core.model.RequestModelActionHandler.executeAction(RequestModelActionHandler.java:1) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.glsp.server.actions.AbstractActionHandler.execute(AbstractActionHandler.java:53) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.glsp.server.internal.actions.DefaultActionDispatcher.runAction(DefaultActionDispatcher.java:198) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.glsp.server.internal.actions.DefaultActionDispatcher.handleAction(DefaultActionDispatcher.java:176) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.glsp.server.internal.actions.DefaultActionDispatcher.handleNextAction(DefaultActionDispatcher.java:163) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.glsp.server.internal.actions.DefaultActionDispatcher.runThread(DefaultActionDispatcher.java:149) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.util.concurrent.ExecutionException: java.lang.RuntimeException: Could not parse 'data' field
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073) ~[?:?]
	at com.eclipsesource.uml.glsp.core.model.UmlModelServerAccess.getSemanticModel(UmlModelServerAccess.java:50) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	... 11 more
Caused by: java.lang.RuntimeException: Could not parse 'data' field
	at org.eclipse.emfcloud.modelserver.internal.client.ModelServerClientDelegate.lambda$37(ModelServerClientDelegate.java:856) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
	at org.eclipse.emfcloud.modelserver.internal.client.ModelServerClientDelegate.require(ModelServerClientDelegate.java:856) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at java.util.function.Function.lambda$andThen$1(Function.java:88) ~[?:?]
	at org.eclipse.emfcloud.modelserver.client.Response.body(Response.java:41) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at org.eclipse.emfcloud.modelserver.internal.client.ModelServerClientDelegate.lambda$6(ModelServerClientDelegate.java:205) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[?:?]
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179) ~[?:?]
	at org.eclipse.emfcloud.modelserver.internal.client.ModelServerClientDelegate$3.onResponse(ModelServerClientDelegate.java:818) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519) ~[com.eclipsesource.uml.cli-0.1.0-SNAPSHOT-standalone.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	... 1 more`

**Comment by haydar-metin** [comment]
Thanks for the report!

I tested the released version against Win10/Linux again and it still works.

I will try the newer java versions the next days and return to you.

Could you provide the path you used? Maybe some kind of character crashes the server.

**Comment by chabanachtaTU** [comment]
Thanks for looking into this. Testfile(s) where at "D:\Privat\Uni\SE\Uebungen\SE_UE1\bigumltestfile.uml", but I also tested in %USERPROFILE%\Documents now.

vscode installation paths are standard, i.e. 
- %USERPROFILE%\AppData\Local\Programs\Microsoft VS Code
- %USERPROFILE%\.vscode\extensions\bigmodelingtools.umldiagram-0.5.1

openjdk is in %PROGRAMFILES%\Java\openjdk...

**Comment by haydar-metin** [comment]
Did you create a new file through `Files -> New File`?

**Comment by chabanachtaTU** [comment]
Thanks - that put me on the right track. It actually was a rather stupid user error. 

After installing the extension, I entered "bigUML: New Empty UML Diagram" in a hurry without an open vscode workspace. VSCode doesn't handle this then through the extension and the file dialogue opens to create a new file with this (useless) filename. Then I chose some other name and thereby didn't create the uml file by using the extension.

Tried it know , it works. Thanks again. Nice tool.
